.clang-tidy

```diff
@@ -6,14 +6,16 @@ Checks:
   modernize-*,
   -modernize-avoid-c-arrays,
   -modernize-use-trailing-return-type,
+  -modernize-concat-nested-namespaces,
   performance-*,
   -performance-avoid-endl
 '
-# WarningsAsErrors: '*'
-HeaderFilterRegex: ''
+WarningsAsErrors:  '*'
+HeaderFilterRegex: '.*'
+ExcludeHeaderFilterRegex: 'include/.*'
 UseColor: true
 FormatStyle: 'file'
-ExtraArgs: ['-std=c++17']
+ExtraArgs: ['-std=c++20']
```

Comment:
We use C++20 types (std::span) for a test in C++20 mode; clang-tidy doesn't like -std=c++17 for that test.